aws/resource_aws_codebuild_project.go

```diff
@@ -683,6 +693,9 @@ func resourceAwsCodeBuildProjectSourceHash(v interface{}) int {
 	if v, ok := m["buildspec"]; ok {
 		buf.WriteString(fmt.Sprintf("%s-", v.(string)))
 	}
+	if v, ok := m["git_clone_depth"]; ok {
+		buf.WriteString(fmt.Sprintf("%d-", v.(int)))
```

Comment:
Similar note here, if `git_clone_depth` defaults to 0, this will change the TypeSet hash when folks upgrade their AWS provider and it will show a diff. We'll need to implement a state migration if that's the case to prevent the diff.